crates/workers/src/workers.rs

```diff
@@ -0,0 +1,571 @@
+            key_storage,
+            scopes,
+        })
+    }
+
+    pub async fn create_worker(
+        &self,
+        deal_id: String,
+        init_peer_id: PeerId,
+    ) -> Result<PeerId, WorkersError> {
+        let worker_id = {
+            let guard = self.worker_ids.read();
+            guard.get(&deal_id).cloned()
+        };
+        match worker_id {
+            Some(_) => Err(WorkersError::WorkerAlreadyExists { deal_id }),
+            _ => {
+                let key_pair = self
+                    .key_storage
+                    .create_key_pair()
+                    .await
+                    .map_err(|err| WorkersError::CreateWorkerKeyPair { err })?;
+
+                let worker_id = key_pair.get_peer_id();
+
+                match self
+                    .store_worker(worker_id, deal_id.clone(), init_peer_id)
+                    .await
+                {
+                    Ok(worker_info) => {
+                        let mut worker_ids = self.worker_ids.write();
+                        let mut worker_infos = self.worker_infos.write();
+
+                        if worker_ids.contains_key(&deal_id) {
+                            return Err(WorkersError::WorkerAlreadyExists { deal_id });
+                        }
+
+                        worker_ids.insert(deal_id, worker_id);
+                        worker_infos.insert(worker_id, worker_info);
+                    }
+                    Err(err) => {
+                        tracing::warn!(
+                            target = "worker-registry",
+                            worker_id = worker_id.to_string(),
+                            "Failed to store worker info for {worker_id}: {}",
+                            err
+                        );
+                        self.key_storage
+                            .remove_key_pair(worker_id)
+                            .await
+                            .map_err(|err| WorkersError::RemoveWorkerKeyPair { err })?;
+
+                        return Err(err);
+                    }
+                }
+
+                Ok(worker_id)
+            }
+        }
+    }
+
+    pub fn get_worker_keypair(&self, worker_id: PeerId) -> Result<KeyPair, WorkersError> {
+        if self.scopes.is_host(worker_id) {
+            Ok(self.key_storage.root_key_pair.clone())
+        } else {
+            self.key_storage
+                .get_key_pair(worker_id)
+                .ok_or(WorkersError::KeypairNotFound(worker_id))
+        }
+    }
+
+    pub fn list_workers(&self) -> Vec<WorkerId> {
+        self.worker_infos.read().keys().cloned().collect()
+    }
+
+    pub fn get_deal_id(&self, worker_id: PeerId) -> Result<DealId, WorkersError> {
+        self.worker_infos
+            .read()
+            .get(&worker_id)
+            .ok_or(WorkersError::WorkerNotFound(worker_id))
+            .map(|info| info.deal_id.clone())
+    }
+    pub async fn remove_worker(&self, worker_id: PeerId) -> Result<(), WorkersError> {
```

Comment:
```suggestion
    }

    pub async fn remove_worker(&self, worker_id: PeerId) -> Result<(), WorkersError> {
```

strange that `cargo fmt` doesn't keep track of that